### ISSUE TITLE
Warn when no version is specified

### ISSUE THREAD

**Issue by ccoulombe** [ComputeCanada/software-stack#122]
## TLDR 
- for a specific set of modules, not all (e.g `scipy-stack`, `python`, `arrow`, `opencv`)
- warn when user load `scipy-stack` without pinning a version

## Goals:
- Decrease the number of users, job crashes and tickets related to updated modules where one did not pin a version and got the latest by default
- Increase reproducibility of job scripts
---

When we add a new version of a module like `scipy-stack`, it can create issues for users who do not specify a version for the module they use. Hence, some users get the latest module without realizing it and their code may break resulting in a job crash and tickets. This also hinders reproducibility of job scripts as the modules loaded may have changed.

I suggest to add a warning to a specific set of modules : `scipy-stack`, `arrow`, `python`, `opencv`, and others we judge useful, when no version is specified . Something like
```
$ module load scipy-stack
Please specify a version instead of relying on the default one! -> module load scipy-stack/2023a.
```
